weather_mv/loader_pipeline/ee.py

```diff
@@ -0,0 +1,371 @@
+        return os.path.join(self.tiff_location, file_name)
+
+    def process(self, uri: str) -> t.Iterator[TiffData]:
+        """Opens grib files and yields TiffData."""
+
+        logger.info(f'Converting {uri!r} to COGs ...')
+        with open_dataset(uri,
+                          self.open_dataset_kwargs,
+                          self.disable_in_memory_copy,
+                          self.disable_grib_schema_normalization) as ds:
+
+            attrs = ds.attrs
+            data = list(ds.values())
+            tiff_name = self._get_tiff_name(uri)
+            start_time, end_time = (attrs.get(key) for key in ('start_time', 'end_time'))
+            dtype, crs, transform = (attrs.pop(key) for key in ['dtype', 'crs', 'transform'])
+
+            file_name = f'{tiff_name}.tiff'
+
+            with tempfile.TemporaryDirectory() as temp_dir:
+                local_tiff_path = os.path.join(temp_dir, file_name)
+                with rasterio.open(local_tiff_path, 'w',
+                                   driver='GTiff',
+                                   dtype=dtype,
+                                   width=data[0].data.shape[1],
+                                   height=data[0].data.shape[0],
+                                   count=len(data),
+                                   crs=crs,
+                                   transform=transform,
+                                   tiled=True) as f:
+                    for i, da in enumerate(data):
+                        f.write(da, i+1)
+
+                # Copy local tiff to gcs.
+                target_path = self._get_target_path(file_name)
+                with open(local_tiff_path, 'rb') as src:
+                    with FileSystems().create(target_path) as dst:
+                        shutil.copyfileobj(src, dst, WRITE_CHUNK_SIZE)
+
+                channel_names = [da.name for da in data]
+                tiff_data = TiffData(
+                        name=tiff_name,
+                        target_path=target_path,
+                        channel_names=channel_names,
+                        start_time=start_time,
+                        end_time=end_time,
+                        properties=attrs
+                    )
+
+                yield tiff_data
+
+
+@dataclasses.dataclass
+class IngestIntoEE(beam.DoFn):
+    """Ingests tiff image into earth engine and yields task id.
+
+    Attributes:
+        ee_asset: The asset folder path in earth engine project where the tiff image files will be pushed.
+        use_personal_account: A flag to authenticate earth engine using personal account. Default: False.
+    """
+
+    ee_asset: str
+    use_personal_account: bool
+
+    @retry.with_exponential_backoff()
+    def start_ingestion_task(self, asset_request: t.Dict) -> str:
+        """Sends a manifest task request to EE for asset ingestion. Returns the task id."""
+        task_id = ee.data.newTaskId(1)[0]
+        _ = ee.data.startIngestion(task_id, asset_request)
```

Comment:
For a COG-backed asset, the call you want here is ee.data.createAsset. That will return immediately and not start a task. See the documentation here: https://developers.google.com/earth-engine/apidocs/ee-data-createasset